### DISCUSSION TITLE
User-defined Trees

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5639] [Feature Requests]
> Is there any plan to add user defined trees to future editions of Specify 7? For example, for aquatic collections (e.g., fishes, inverts, etc.), **it would be extremely beneficial to be able to include information about drainages in the Collecting Event information.** 
>
>Like Geography, drainages are arranged hierarchically and would be best organized into a tree format. For data analyses and mining information from collection records, drainage information is more valuable and useful than geography (fishes don’t pay attention to county and state lines,). It would be great to be able to search river systems or watersheds for collection records.
> 
> Pre-determined Drainage Trees designed by Specify programmers are not necessary. If users could just have a blank Tree that they could design themselves, that would be amazing. 
>
>Some users might want to use it for drainage, others may have other specific types of data they wish to include that lend themselves to the tree format.

**Requested By:** Sarah at VIMS

**Comment by grantfitzsimmons** [comment]
[Community Forum Post](https://discourse.specifysoftware.org/t/user-defined-trees/726/3)

**Niels Klazenga at Royal Botanic Gardens Victoria**

> Yes, great idea. I would not mind having an extra geography tree for example for the TDWG World Geographic Scheme for Recording Plant Distributions ([World Geographical Scheme for Recording Plant Distributions (WGSRPD) - TDWG](https://www.tdwg.org/standards/wgsrpd/); plants do not really care about country borders either).
> 
> For maximum usability, it would be good if the rank (or TreeDefItem) could be optional, but that might get too complicated and a better way for this might be to allow hierarchical pick lists.

**Comment by maxpatiiuk** [comment]
Way smaller, but as part of this the trees dialog would have to be improved (because there would be more than 3 tries).

I.e, ability to reorder the trees or group them somehow

**Comment by grantfitzsimmons** [comment]
This has come up again for drainage trees 

**Requested By:**
> Gabriela M. Hogue
> Collections Manager, Ichthyology
> North Carolina Museum of Natural Sciences

**Comment by grantfitzsimmons** [comment]
Drainage tree.

**Requested By:** KU Ichthyology

**Comment by grantfitzsimmons** [comment]
https://github.com/specify/specify7/blob/312139ad3ee9290dd00500e764dd0f0c36f8b026/specifyweb/frontend/js_src/lib/components/InitialContext/treeRanks.ts#L34-L47

Sadly, the trees that display in the dialog are determined by this logic instead of being user-definable. This means I can't add a custom drainage tree for @acbentley unless the `type` of discipline is set to 'paleobotany', 'invertpaleo', or 'vertpaleo'.

**Comment by maxpatiiuk** [comment]
At one point I remember Theresa requesting we add a global pref that enables those trees in non-paleo disciplines too.
I don't think it was implemented - shouldn't be hard (besides testing effort)
However I would recommend implementing it only once global preferences are reimplemented in sp7 to diverse from sp6

**Comment by grantfitzsimmons** [comment]
In addition to a drainage tree, **PRI Paleo** would like to see a Ichno tree:

>Ichnotaxonomy is a branch of paleontology that focuses on the classification and study of trace fossils, which are traces or remains of the activity of ancient organisms. It involves identifying and categorizing trace fossils based on their size, shape, structure, and other characteristics. Ichnotaxonomy provides valuable information about the behavior, ecology, and evolutionary history of organisms that lived in the past, even if their actual bodies are not preserved in the fossil record.

>There's a potential grant for HaptoNet, focusing on organisms that live on other organisms (traces of things).
For example, a barnacle trace left by a barnacle.
>Ichno taxonomy covers both fossils and modern traces.
>When a trace is assigned a Latin binomial, it lacks a way to indicate that it's a trace of that organism.
>We can publish the information about feeding traces on VertNet.
>This will contribute to the ichno taxonomy tree.

**Comment by g1rly-c0d3r** [comment]
linked in https://github.com/specify/specify7/issues/8217
